### DISCUSSION TITLE
How to add new or custom notification features like 'comment posted', 'mentioned'...

### DISCUSSION THREAD

**Discussion by namnguyenthanhwork** [outline/outline#7143] [Self Hosting]
I want to add new feature in notification tab and sending emails when active it.

First, I add new notification feature. 

I add new `NotificationEventType` in `shared/types.ts`

```
DailyDigestUpdatedDocument = "documents.daily_digest_updated",
```

and set default `true` for it

```
[NotificationEventType.DailyDigestUpdatedDocument]: true,
```

Second, I add new option in `app/scenes/Settings/Notifications.tsx`

```
 {
      event: NotificationEventType.DailyDigestUpdatedDocument,
      icon: <StarredIcon />,
      title: t("Daily digest updated documents"),
      description: t("Receive daily notifications a list of updated documents"),
    },
```

<img width="993" alt="image" src="https://github.com/outline/outline/assets/78526193/9617fb71-361e-4226-b579-6280cdbf7177">

Third, I create new case in `server/queues/processors/EmailsProcessor.ts` (I copy logic MentionedInDocument)

```
  case NotificationEventType.DailyDigestUpdatedDocument: {
        // No need to delay email here as the notification itself is already delayed
        await new DocumentUpdatedDailyDigest(
          {
            to: notification.user.email,
            documentId: notification.documentId,
            teamUrl: notification.team.url,
            actorName: notification.actor.name,
          },
          { notificationId }
        ).schedule();
        return;
      }
```

Finally, I copy logic `server/emails/templates/DocumentMentionedEmail.tsx` and change class name to `DocumentUpdatedDailyDigest` to test send mail. But, it doesn't work.

Maybe I loss some logic code. Can anyone help me see if I'm missing anything? Thanks.

**Comment by tommoor** [comment]
Looks like you will still need to create the event `NotificationEventType.DailyDigestUpdatedDocument` that the processor acts upon. You need something that pushes these into the event queue, I'd probably create a scheduled task – look at `CleanupDeletedDocumentsTask` as an example that runs daily

**Reply by namnguyenthanhwork** [reply]
I tried the suggestion you recommended, but it's not working. Since I'm new to this project, could you suggestion for me some steps to create this feature? Thanks for your help! @tommoor 

Idea for this feature: The system will email team members daily a list of newly created documents that day.
